server-common/src/test/java/org/apache/kafka/server/util/timer/MockTimer.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.server.util.timer;
+
+import org.apache.kafka.server.util.MockTime;
+
+import java.util.Comparator;
+import java.util.PriorityQueue;
+
+public class MockTimer implements Timer {
+    private final PriorityQueue<TimerTaskEntry> taskQueue = new PriorityQueue<>(
+        Comparator.comparingLong(entry -> entry.expirationMs)
+    );
+
+    public final MockTime time;
+
+    public MockTimer() {
+        this(new MockTime());
+    }
+
+    public MockTimer(MockTime time) {
+        this.time = time;
+    }
+
+    @Override
+    public void add(TimerTask timerTask) {
+        if (timerTask.delayMs <= 0) {
+            timerTask.run();
+        } else {
+            synchronized (taskQueue) {
+                taskQueue.add(new TimerTaskEntry(timerTask, timerTask.delayMs + time.milliseconds()));
```

Comment:
please check the return value for `add()`. If false, you might want to handle it appropriately. I know that scala code also didn't do it so it would be ok if you create a JIRA for it.

Reply:
As per the [javadoc](https://docs.oracle.com/javase/8/docs/api/java/util/PriorityQueue.html#add-E-) of `PriorityQueue`, `add` always return `true` so we don't need to do this.